### ISSUE TITLE
mask rcnn 转化

### ISSUE THREAD

**Issue by careolesff** [sophgo/tpu-mlir#234]
在使用官方tpu-mlir的maskrcnn转化教程进行detectron2框架下的模型转化时出现了 Assertion `roi_len == 5' failed.报错。
问题在于tpuc-opt: /workspace/tpu-mlir/lib/Dialect/Top/Interfaces/MaskRCNNGetBboxB.cpp:31: void tpu_mlir::top::MaskRCNNGetBboxBOp::shape_inference(): Assertion `roi_len == 5' failed
我的配置文件如下：
# MaskRCNN big-op config (with p6 in backbone/RPN, ROI 仅使用 p2-p5)
# Classes: 25

# ===== Channel & structural parameters =====
CHANNEL_ROI: 256
CHANNEL_RPN_SCORES: 3            # anchors per location
CHANNEL_RPN_BBOXES: 12           # 4 * anchors
NUM_LEVELS: 5                    # p2..p6 for backbone & RPN
NUM_LEVELS_ROI: 4                # ROI pooler 仍只用 p2..p5
GLOBAL_BATCH_SIZE: 1
NUM_CLASSES: 25
NUM_CLASSES_GetBboxB: 25
Global_strides_DQ:               # 每个 FPN level 的 (stride_h, stride_w)；用于解码/量化
  - [4,4]    # p2
  - [8,8]    # p3
  - [16,16]  # p4
  - [32,32]  # p5
  - [64,64]  # p6

# ===== ROI / proposal sizing =====
ROI_PH_BBOX_POOLER: 7
ROI_PW_BBOX_POOLER: 7
ROI_PH_MASK_POOLER: 14
ROI_PW_MASK_POOLER: 14
ROI_SLICE_BBOX_POOLER: 250       # Max pooled proposals for box head
ROI_SLICE_MASK_POOLER: 100       # Max pooled proposals for mask head
MAX_PER_IMG: 250
MAX_PER_IMG_GetBboxB: 100
ROI_LEN: 5
NUM_INDEXES: 4

# ===== Image / feature spatial constants =====
DELTA2BBOX_1st_MAX_SHAPE_H: 800.0
DELTA2BBOX_1st_MAX_SHAPE_W: 1216.0
DELTA2BBOX_2nd_MAX_SHAPE_H: 800.0
DELTA2BBOX_2nd_MAX_SHAPE_W: 1216.0
H_RPN_DYN_MAX: 200               # p2 height
W_RPN_DYN_MAX: 304               # p2 width
ROI_H: 200
ROI_W: 304

# ===== Normalization / bbox delta stats (example, adapt if you trained with different stats) =====
DELTA2BBOX_1st_MEAN: 0
DELTA2BBOX_1st_STD_0: 1
DELTA2BBOX_1st_STD_1: 1
DELTA2BBOX_2nd_MEAN: 0
DELTA2BBOX_2nd_STD_0: 0.1
DELTA2BBOX_2nd_STD_1: 0.2

# ===== Thresholds & NMS =====
CONF_THRESHOLD_1st: 0.1
CONF_THRESHOLD_2nd: 0.0
GetBboxB_SCORE_EQ: 0.05
NMS_THRE_1st: 0.7
NMS_THRE_2nd: 0.5
NMS_PRE: 1000
NMS_SLICE: 1000
TOPK_ONNX_NMS_1st: 250
TOPK_ONNX_NMS_2nd: 250

# ===== Dynamic length / hardware tuning =====
FAKE_DQ_DYNAMIC_LENGTH: 4741
MAX_LENGTH_STATIC_STRECHED: 4741
NMS_MAX_LENGTH_1st: 4741
NMS_MAX_LENGTH_2nd: 20000
MASK_POOLER_FAKE_DYNAMIC_LENGTH: 100
HARDWARE_FACTOR_TOPK: 10
TPU1686_TOPK_FACTOR: 3
MAX_SCALAR_C: 4.1351665567

# ===== Big-op external params =====
maskrcnn_model_transform_extern_params:
  # Module block types: 1(torch) 0(PPL op): backbone -> RPN -> Bbox_Pooler -> bbox head -> get_bboxes_B -> mask pooler -> mask head
  list_encoded_block: 1,0,0,1,0,0,1
  list_name_PPLop: ppl::RPN_get_bboxes,ppl::Bbox_Pooler,ppl::get_bboxes_B,ppl::Mask_Pooler
  # Backbone outputs (15):
  # 0:feat_p2 1:feat_p3 2:feat_p4 3:feat_p5 4:feat_p6
  # 5:rpn_cls_p2 6:rpn_cls_p3 7:rpn_cls_p4 8:rpn_cls_p5 9:rpn_cls_p6
  # 10:rpn_bbox_p2 11:rpn_bbox_p3 12:rpn_bbox_p4 13:rpn_bbox_p5 14:rpn_bbox_p6
  # RPN_get_bboxes consumes 10 logits + max_shape => 11 inputs
  maskrcnn_io_map: '{(0,0):(-1,0),(1,0):(0,5),(1,1):(0,6),(1,2):(0,7),(1,3):(0,8),(1,4):(0,9),(1,5):(0,10),(1,6):(0,11),(1,7):(0,12),(1,8):(0,13),(1,9):(0,14),(1,10):(-1,1),(2,0):(0,0),(2,1):(0,1),(2,2):(0,2),(2,3):(0,3),(2,4):(1,0),(3,0):(2,0),(4,0):(2,1),(4,1):(3,1),(4,2):(3,0),(4,3):(-1,2),(4,4):(-1,3),(5,0):(0,0),(5,1):(0,1),(5,2):(0,2),(5,3):(0,3),(5,4):(4,0),(5,5):(4,1),(5,6):(-1,4),(6,0):(5,0),(-2,0):(4,0),(-2,1):(4,1),(-2,2):(6,0)}'
  maskrcnn_num_output: '3'
  # Inputs per PPLOp (RPN_get_bboxes=11, Bbox_Pooler=5, get_bboxes_B=5, Mask_Pooler=7)
  nums_PPLOp_Inputs: 11,5,5,7

算子的拓扑结构与官方给出的图片完全一致
